### PR TITLE
Add caliper format and code for edx.course.enrollment.mode_changed event

### PR DESCRIPTION
**Pull request by HamzaFarooqArbisoft**
**Trello Link:** _https://trello.com/c/VE0jW7iS/55-enrollment-event-edxcourseenrollmentmodechanged_

**Description:** _Add caliper format and code for edx.course.enrollment.mode_changed_

**Checks before merge:**

- [ ] Reviewed
- [x] Commits squashed

### CODE REVIEW

**Comment by asamolion** [comment]
@arooshaarif Please review.

File: openedx/features/caliper_tracking/tests/expected/edx.course.enrollment.mode_changed.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      "id": "http://localhost:18000/courses/course-v1:PUCIT+CS520+2018_T4/about",
+      "type": "Membership"
+   },
+   "eventTime":"2018-11-16T07:52:15.893Z",
+
+   "referrer": {
+      "id": "http://localhost:18000/courses/course-v1:PUCIT+CS520+2018_T4/about",
+      "type": "WebPage"
+   },
+
```
Comment (arooshaarif) [inline openedx/features/caliper_tracking/tests/expected/edx.course.enrollment.mode_changed.json]: Remove extra spaces.

Comment (HamzaFarooqArbisoft) [reply openedx/features/caliper_tracking/tests/expected/edx.course.enrollment.mode_changed.json]: Done

File: openedx/features/caliper_tracking/tests/expected/edx.course.enrollment.mode_changed.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "id": "http://localhost:18000/u/honor",
+    "name": "honor",
+    "type": "Person"
+  },
+  "eventTime": "2018-11-16T07:52:15.893Z",
+  "extensions": {
+    "extra_fields": {
+      "accept_language": "en-US,en;q=0.9",
+      "agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36",
+      "course_id": "course-v1:PUCIT+CS520+2018_T4",
```
Comment (asamolion) [inline openedx/features/caliper_tracking/tests/expected/edx.course.enrollment.mode_changed.json]: I believe `course_id` and `org_id` should be moved inside `extensions` key of `object`

Comment (HamzaFarooqArbisoft) [reply openedx/features/caliper_tracking/tests/expected/edx.course.enrollment.mode_changed.json]: Done

File: openedx/features/caliper_tracking/transformers/enrollment_transformers.py [hunk 3]
```diff
@@ -24,7 +24,8 @@ def edx_course_enrollment_activated(current_event, caliper_event):
             'extensions': {
                 'mode': current_event['event']['mode'],
                 'course_id': current_event['context']['course_id'],
-                'user_id': caliper_event['extensions']['extra_fields'].pop('user_id')
+                'user_id': caliper_event['extensions']['extra_fields'].pop(
+                    'user_id')
```
Comment (asamolion) [inline openedx/features/caliper_tracking/transformers/enrollment_transformers.py]: What is this ?

Comment (HamzaFarooqArbisoft) [reply openedx/features/caliper_tracking/transformers/enrollment_transformers.py]: `user_id` was placed in `caliper_event` in `base_transformer` but i moved `user_id` to `extensions of object`, this is why i had to place and pop `user_id`, which i preferred to be done simultaneously.

Comment (asamolion) [reply openedx/features/caliper_tracking/transformers/enrollment_transformers.py]: I mean why have you indented `'user_id')` like that?

Comment (HamzaFarooqArbisoft) [reply openedx/features/caliper_tracking/transformers/enrollment_transformers.py]: It's due to pep8 issue.

**Pull request merged**